### PR TITLE
Eth block tracker

### PR DESCRIPTION
**Pull request by darkdh**
Resolves https://github.com/brave/brave-browser/issues/17009

The tracker can query block number on certain network according to the interval. It can also be stopped when we need to save bandwidth. Also provide an API to query block number directly without waiting for periodically polling.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/browser/brave_wallet_service.cc [hunk 1]
```diff
@@ -36,6 +37,11 @@ BraveWalletService::BraveWalletService(
       std::make_unique<brave_wallet::AssetRatioController>(url_loader_factory);
   swap_controller_ =
       std::make_unique<brave_wallet::SwapController>(url_loader_factory);
+  block_tracker_ =
+      std::make_unique<brave_wallet::EthBlockTracker>(rpc_controller_.get());
+  // TODO(darkdh): Stop the tracker to save bandwidth when users are not using
+  // wallet actively
+  block_tracker_->Start(base::TimeDelta::FromSeconds(15));
```
Comment (bbondy) [inline components/brave_wallet/browser/brave_wallet_service.cc:44]: Maybe we shouldn't start it yet until we need to use it for something? 
E.g. I think maybe only when a WebUI is open

Comment (bbondy) [reply components/brave_wallet/browser/brave_wallet_service.cc:44]: With our current wallet we only start the block tracker when the UI is open. And close it even if the tab is not active. And we lazy load the extension, and traffic is already very high even with that restricted way.

Comment (darkdh) [reply components/brave_wallet/browser/brave_wallet_service.cc:44]: I will remove it for now since we are not really using it

Comment (darkdh) [reply components/brave_wallet/browser/brave_wallet_service.cc]: removed

**Review by bbondy — APPROVED** [review]

**Pull request merged**